src/SerializeAST.cc

```diff
@@ -754,10 +774,45 @@ bool IsElementResourceGroup(const Element& element)
     return element.element == Element::CategoryElement && element.category == Element::ResourceGroupCategory;
 }
 
+
+#if _WITH_REFRACT_
+typedef std::vector<const snowcrash::DataStructure*> DataStructures;
+
+void findNamedDataStructures(const snowcrash::Elements& elements, DataStructures& found) {
+    for (snowcrash::Elements::const_iterator i = elements.begin()
+       ; i != elements.end()
+       ; ++i) {
+
+        if (i->element == snowcrash::Element::DataStructureElement) {
+            found.push_back(&(i->content.dataStructure));
+        }
+        else if (!i->content.resource.attributes.empty()) {
+            found.push_back(&i->content.resource.attributes);
```

Comment:
I discuss it with @zdne while implement searching for named Types.

Quoted && translated (originally in Czech:)
Named types comes from two sources
first:
```
# R [/r]
+ attributes 
    + p
```
second:
```
# Data Structures
## R
+ p
```

did we miss something?

Will add fixture as you suggest